content/en/docs/advanced-tasks/unsafe-fail-forward-upgrades.md

```diff
@@ -0,0 +1,123 @@
+
+> BEST PRACTICE NOTE: If a bundle is known to fail, it should be skipped in the upgrade graph using the [skips](https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/#skips) or [skipRange](https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/#skiprange) feature.
+
+To further understand, let's visualize this.
+
+{{<mermaid>}}
+graph TD;
+    A(Operavtor V1 installed) --> B(Starts upgrading to Operator V2)
+
+    B --> |Install succeeds| F(Operator V2 is installed and succeeds)
+
+    B --> |InstallPlan fails| C(Operator V3 added to catalog replacing V2)
+    C --> |Upgrade blocked as V1 is in solution set still| D(Manually delete Operator V2 InstallPlan)
+    D --> |New InstallPlan generated and approved| E(Operator V3 is installed and succeeds)
+
+    B --> |InstallPlan fails with UnsafeFailForward upgrades| H(Operator V3 added to catalog replacing V2)
+    H --> |New InstallPlan generated and approved| J(Operator V3 is installed and succeeds)
+{{</mermaid>}}
+
+### When one or more CSVs have failed
+
+Let's review what steps must be taken to recover from a failed CSV today:
+
+- `Operator v1` is being upgraded to `Operator v2`.
+- The CSV for `Operator v2` enters the FAILED phase.
+- `Operator v3` is added to the catalog which replaces or skips `Operator v2`.
+- The resolver cannot upgrade `Operator v2` while including `Operator v1` in the solution set, upgrade is blocked.
+- User manually deletes the existing CSVs, a new InstallPlan is generated and approved.
+- `Operator v3` is installed and the upgrade succeeds.
+
+To further understand, let's visualize this.
+
+{{<mermaid>}}
+graph TD;
+  A(Operator V1 installed) --> B(Starts upgrading to Operator V2)
+
+  B --> |Install succeeds| F(Operator V2 is installed and succeeds)
+
+  B --> |CSV enters FAILED phase| C(Operator V3 added to catalog replacing V2)
+  C --> |Upgrade blocked as V1 is in solution set still| D(Manually delete Operator V2 CSV)
+  D --> |New InstallPlan generated and approved| E(Operator V3 is installed and succeeds)
+
+  B --> |CSV enters FAILED phase with UnsafeFailForward upgrades| H(Operator V3 added to catalog replacing V2)
+  H --> |New InstallPlan generated and approved| J(Operator V3 is installed and succeeds)
+{{</mermaid>}}
+
+When you opt into "UnsafeFailForward" upgrades, OLM:
+- Does not include CSVs in the REPLACING phase in the set of arguments sent to the Resolver if the final CSV in the upgrade chain is in the FAILED phase and all other CSVs are in the REPLACING phase.
+- Allows CSVs to move from the FAILED phase to the REPLACING phase.
+
+These changes allow OLM to recover from any number of failed CSVs in an upgrade path. You only need to update the CatalogSource with new operators that replace the latest FAILED CSV. With catalog polling enabled, you can even skip this step by pushing a new catalog image to the same tag.
```

Comment:
```suggestion
These changes allow OLM to recover from any number of failed CSVs in an upgrade path, replacing the latest FAILED CSV with the next upgrade.
```

Reply:
This goes reverts this comment from Dan:

https://github.com/operator-framework/olm-docs/pull/235#discussion_r850720049

Thoughts?

Reply:
An update to the catalog isn't required to move past failed CSVs, only failed InstallPlans. If the failed CSV was at the head of the upgrade graph, then yes you would need to update the catalog with an upgrade, but I don't want users to think that they need to do this step in both cases.

Reply:
This note could certainly go in the `When an InstallPlan failed` section.